File: frontend/src/api/staff.ts
import { apiClient } from "./client";

export interface Staff {
  id: number;
  full_name: string;
  position: string;
  phone?: string;
  email?: string;
  address?: string;
  salary_type: string;
  hourly_rate?: number;
  package_salary?: number;
  is_active: boolean;
}

export const fetchStaff = async (): Promise<Staff[]> => {
  const response = await apiClient.get("/staff");
  return response.data;
};

export const createStaff = async (data: Partial<Staff>) => {
  const response = await apiClient.post("/staff", data);
  return response.data;
};

export const updateStaff = async (id: number, data: Partial<Staff>) => {
  const response = await apiClient.put(`/staff/${id}`, data);
  return response.data;
};

export const deleteStaff = async (id: number) => {
  const response = await apiClient.delete(`/staff/${id}`);
  return response.data;
};

File: backend/app/models/staff.py
from sqlalchemy import Column, String, Float, Boolean
from app.models.base import TenantBase


class Staff(TenantBase):
    __tablename__ = "staff"

    full_name = Column(String, nullable=False)
    position = Column(String, nullable=False)

    salary_type = Column(String, nullable=False)  # hourly | package
    hourly_rate = Column(Float, nullable=True)
    package_salary = Column(Float, nullable=True)

    phone = Column(String, nullable=True)
    email = Column(String, nullable=True)

    is_active = Column(Boolean, default=True)